src/main/java/net/dries007/tfc/api/capability/metal/CapabilityMetalItem.java

```diff
@@ -0,0 +1,124 @@
+import net.minecraftforge.oredict.OreDictionary;
+
+import net.dries007.tfc.api.capability.DumbStorage;
+import net.dries007.tfc.api.registries.TFCRegistries;
+import net.dries007.tfc.api.types.Metal;
+import net.dries007.tfc.api.util.TFCConstants;
+import net.dries007.tfc.objects.inventory.ingredient.IIngredient;
+import net.dries007.tfc.util.Helpers;
+
+public final class CapabilityMetalItem
+{
+    @CapabilityInject(IMetalItem.class)
+    public static final Capability<IMetalItem> METAL_OBJECT_CAPABILITY = Helpers.getNull();
+    public static final ResourceLocation KEY = new ResourceLocation(TFCConstants.MOD_ID, "metal_object");
+
+    public static final Map<IIngredient<ItemStack>, Supplier<ICapabilityProvider>> CUSTOM_ITEMS = new HashMap<>(); //Used inside CT, set custom IItemSize for items outside TFC
+
+    public static void preInit()
+    {
+        CapabilityManager.INSTANCE.register(IMetalItem.class, new DumbStorage<>(), MetalItemHandler::new);
+    }
+
+    /**
+     * Gets the IMetalItem instance from an itemstack, either via capability or via interface
+     *
+     * @param stack The stack
+     * @return The IMetalItem if it exists, or null if it doesn't
+     */
+    @Nullable
+    public static IMetalItem getMetalItem(ItemStack stack)
+    {
+        if (!stack.isEmpty())
+        {
+            if (stack.getItem() instanceof IMetalItem)
+            {
+                return (IMetalItem) stack.getItem();
+            }
+            else if (stack.getItem() instanceof ItemBlock && ((ItemBlock) stack.getItem()).getBlock() instanceof IMetalItem)
+            {
+                return (IMetalItem) ((ItemBlock) stack.getItem()).getBlock();
+            }
+            return stack.getCapability(METAL_OBJECT_CAPABILITY, null);
+        }
+        return null;
+    }
+
+    @Nullable
+    public static ICapabilityProvider getCustomMetalItem(ItemStack stack)
+    {
+        if (!stack.isEmpty())
+        {
+            Set<IIngredient<ItemStack>> itemItemSet = CUSTOM_ITEMS.keySet();
+            for (IIngredient<ItemStack> ingredient : itemItemSet)
+            {
+                if (ingredient.testIgnoreCount(stack))
+                {
+                    return CUSTOM_ITEMS.get(ingredient).get();
+                }
+            }
+            //Try using ore dict prefix-suffix common values (ie: ingotCopper)
+            int[] ids = OreDictionary.getOreIDs(stack);
+            for (int id : ids)
+            {
+                ICapabilityProvider handler = getMetalItemFromOreDict(OreDictionary.getOreName(id));
+                if (handler != null)
+                {
+                    return handler;
+                }
+            }
+        }
+        return null;
+    }
+
+    @Nullable
+    private static ICapabilityProvider getMetalItemFromOreDict(String oreDict)
+    {
+        Metal.ItemType type;
+        String remaining;
+        int amount;
+        if (oreDict.startsWith("ingot"))
```

Comment:
Instead of this, how about holding a `Map<String, Metal.ItemType>`:
During `preInit`, populate the map with entries as such:
```java
ORE_DICT_METAL_ITEMS.add("ingot", Metal.ItemType.INGOT);
// include nugget, double ingot, sheet, double sheet, scrap, dust
```
Then when you need to find a matching item:
```java
for (String oreName : ORE_DICT_METAL_ITEMS.keySet()) {
    if (oreDict.startsWith(oreName)) {
        type = ORE_DICT_METAL_ITEMS.get(oreName);
        remaining = oreDict.substring(oreName.length());
        break;
    }
}
// get the amount by later calling type.getSmeltAmount();
```